rancher/provider.go

```diff
@@ -80,29 +98,37 @@ func init() {
 
 		"secret_key": "API secret used to authenticate with the rancher server",
 
-		"api_url": "The URL to the rancher API, must include version uri (ie. v1 or v2-beta)",
+		"api_url": "The URL to the rancher API",
+
+		"cattle_access_key": "API Key used to authenticate with the rancher server",
+
+		"cattle_secret_key": "API secret used to authenticate with the rancher server",
+
+		"cattle_url": "The URL to the rancher API",
 
 		"config": "Path to the Rancher client cli.json config file",
 	}
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	// Fetch configuration from deprecated environment variables
 	apiURL := d.Get("api_url").(string)
 	accessKey := d.Get("access_key").(string)
 	secretKey := d.Get("secret_key").(string)
 
+	// Override with new environment variables
+	apiURL = d.Get("cattle_url").(string)
+	accessKey = d.Get("cattle_access_key").(string)
+	secretKey = d.Get("cattle_secret_key").(string)
```

Comment:
Why duplicate the keys if they have exactly the same role?

Reply:
So that we can use the environment variables provided by cattle

Reply:
Oh I see…